pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateData.java

```diff
@@ -34,7 +34,7 @@ public record ServiceUnitStateData(
 
     public ServiceUnitStateData {
         Objects.requireNonNull(state);
-        if (StringUtils.isBlank(dstBroker) && StringUtils.isBlank(sourceBroker)) {
```

Comment:
The intention is that we want to tag a source or destination broker for all events to trace where the event comes from.

For example, the logic tries to set "source broker" when setting Free state, to indicate who was the previous owner. In what cases, both source and dst brokers can be empty here?

Reply:
https://github.com/apache/pulsar/blob/b1c5d96ab480ba1b8ad8cbe2077cbe4c467dfc61/pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java#L1364

If the state is not `Owned`, there is still a case that `orphanData.sourceBroker()` is null. Here is the log I debugged

```
ServiceUnitStateData[state=Assigning, dstBroker=localhost:50930, sourceBroker=null,
```